R/join_keys.R

```diff
@@ -0,0 +1,604 @@
+# Constructors ====
+
+#' Create a `JoinKeys` out of a list of `JoinKeySet` objects
+#'
+#' @description `r lifecycle::badge("stable")`
+#'
+#' @details Note that join keys are symmetric although the relationship only needs
+#' to be specified once.
+#'
+#' @name join_keys
+#'
+#' @param ... optional, a `JoinKeySet` objects created using the `join_key` function.
+#'
+#' @return `JoinKeys`
```

Comment:
I was planning on keeping it as we are not keeping them both at the same time.

However a cleanup/standardization to snake_case would be nice. WDYT?

Reply:
Since we now have classes like `teal_slice` and `teal_data` I think snake case would fit better.
We could also go with `join_key/join_keys` instead of `join_key_set/join_keys` but I don't have a strong opinion on this.

Reply:
`join_key_set` is very easy to change as it only really lives on `{teal.data}`

`JoinKeys` is a bit trickier but I think it's doable (maybe only now as we're breaking so many things) :sweat_smile: 

Done in 5e8ef7eb _(passes all tests locally)_